src/app/core/duplicate-records/duplicate-records.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { EntityMapperService } from '../entity/entity-mapper/entity-mapper.service'; 
+import { EntityRegistry } from '../entity/database-entity.decorator';
+import { EntitySchemaService } from '../entity/schema/entity-schema.service';
+import { Entity } from '../entity/model/entity';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class DuplicateRecordService {
+  get: jasmine.Spy<jasmine.Func>;
+  constructor(
+    private entitymapperservice: EntityMapperService,
+    private entityTypes: EntityRegistry,
+    private entityService: EntitySchemaService,
+  ) {}
+
+  async duplicateRecord(sourceData: Entity[], schemaName: string)  {
+    const duplicateData = this.clone(sourceData, schemaName);
+    return await this.entitymapperservice.saveAll(duplicateData);
+  }
+
+  clone(sourceData: Entity[], schemaName: string): any {
+    const duplicateData = [];
+    const entityConstructor = this.entityTypes.get(schemaName);
```

Comment:
You can get the entity constructor easily even without passing the schemaName as parameter, from the entity instances:

`sourceData[i].getConstructor()` returns the constructor with its schema of the current item. As we only have a few items being cloned usually, I think it won't hurt to move this inside the loop and simply use `item.getConstructor()`. We can even remove the schemaName parameter from the duplicateRecord method then.

Reply:
@sleidig Updated it.